.github/instructions/stream-indicators.instructions.md

```diff
@@ -0,0 +1,367 @@
+---
+applyTo: "src/**/*.StreamHub.cs,tests/**/*.StreamHub.Tests.cs"
+description: "Stream-style indicator development and testing guidelines"
+---
+
+# Stream indicator development guidelines
+
+These instructions apply to stream-style indicators that support real-time data processing with stateful operations. Stream indicators maintain internal state and can process individual quotes as they arrive.
+
+## Stream Hub I/O Scenarios
+
+The codebase implements several types of stream hub I/O patterns:
+
+1. **IQuote → IReusable** (e.g., EMA, SMA): Takes quote input, produces single reusable value output
+2. **IQuote → ISeries** (e.g., Alligator, AtrStop): Takes quote input, produces multi-value series output  
+3. **IReusable → IReusable** (e.g., chained indicators): Takes reusable input, produces reusable output
+4. **IQuote → QuotePart** (e.g., QuotePart selector): Takes quote input, extracts specific price component
```

Comment:
`IQuote → QuotePart` is the one I meant to say is rare, but the one you isolated is also rare.